cclient/api/extended/promote_transaction.c

```diff
@@ -119,8 +119,25 @@ retcode_t iota_client_promote_transaction(iota_client_service_t const* const ser
                 consistency_res->info->data);
   }
 
-  // TODO: return out_bundle
-  (void)out_bundle;
+  if (out_bundle == NULL) {
+    ret_code = RC_NULL_PARAM;
+    log_error(client_extended_logger_id, "%s the out_bundle cannot be NULL \n", __func__, error_2_string(ret_code));
+    goto done;
+  }
+  flex_trit_t* array_elt = NULL;
+  iota_transaction_t tx = {};
+  int tx_deserialize_offset = 0;
+  HASH_ARRAY_FOREACH(att_res->trytes, array_elt) {
+    tx_deserialize_offset = transaction_deserialize_from_trits(&tx, array_elt, true);
```

Comment:
`transaction_deserialize_from_trits` return `size_t`, but `tx_deserialize_offset` is `int` and it can be removed since it's meaningless.
```suggestion
    if( transaction_deserialize_from_trits(&tx, array_elt, true) == NUM_TRITS_SERIALIZED_TRANSACTION) {
```